cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/config/ETLBatchConfig.java

```diff
@@ -29,34 +31,49 @@
 public final class ETLBatchConfig extends ETLConfig {
   private final String schedule;
   private final List<ETLStage> actions;
+  private final List<LocalizeResourceInfo> resourcesToLocalize;
```

Comment:
Lets say that a pipeline has the following:

1. Source: File  - emits a blob of text
2. Transform1: Parser - parses data based
3. Transform2: Business Logic (for lack of a better phrase), but lets say it does some transformations other than the Parser
4. Sink1: Sink for Type1 data
5. Sink2: Sink for Type2 data

The pipeline uses a properties file that is shared across these plugins. Each plugin makes decisions on incoming data based on different sections (sets of keys) in the properties file.

In this case, if the plugins are independent, then every plugin that needs the properties file would have to localize a copy for itself.

I thought instead it would be nice to have the pipeline itself do the localization, and the plugins to access the same localized file.